bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -24,14 +27,31 @@ def script_compiler
     @script_compiler ||= Puppet::Pal::ScriptCompiler.new(closure_scope.compiler)
   end
 
-  def run_task(executor, targets, name, args = {})
+  def inventory
+    Puppet.lookup(:bolt_inventory)
+  end
+
+  def get_task(name)
     tasksig = script_compiler.task_signature(name)
     raise Bolt::Error.new("#{name} could not be found", 'bolt/apply-prep') unless tasksig
+    tasksig
+  end
+
+  # rubocop:disable Naming/AccessorMethodName
+  def set_agent_feature(target)
+    inventory.set_feature(target, 'puppet-agent')
+  end
+  # rubocop:enable Naming/AccessorMethodName
+
+  def run_task(targets, tasksig, args = {})
+    errors = []
+    unless tasksig.runnable_with?(args || {}) { |msg| errors << msg }
```

Comment:
This is puppety and should probably happen outside of the concurrent section for clarity.